src/modules/user/user.service.ts

```diff
@@ -233,4 +234,32 @@ export default class UserService {
       },
     };
   }
+
+  async softDeleteUser(userId: string, authenticatedUserId: string): Promise<any> {
+    const identifierOptions: UserIdentifierOptionsType = {
+      identifier: userId,
+      identifierType: 'id',
+    };
+
+    const user = await this.getUserRecord(identifierOptions);
+
+    if (!user) {
+      throw new NotFoundException('User not found');
```

Comment:
Meet cobidex on the custom exception handler he created

Reply:
@kaleabendrias